src/main/java/org/wso2/carbon/connector/FileAppend.java

```diff
@@ -103,20 +103,22 @@ private boolean appendFile(String destination, String content, String position,
                 fileObj.createFile();
             }
             reader = new BufferedReader(new InputStreamReader(fileObj.getContent().getInputStream()));
-            List<String> lines = reader.lines().collect(Collectors.toList());
-            if (StringUtils.isNotEmpty(position) && Integer.parseInt(position) <= lines.size()
-                    && Integer.parseInt(position) > 0) {
-                lines.add(Integer.parseInt(position) - 1, content);
-                out = fileObj.getContent().getOutputStream();
-                if(StringUtils.isEmpty(encoding)) {
-                    IOUtils.writeLines(lines, null, out, DEFAULT_ENCODING);
-                } else {
-                    IOUtils.writeLines(lines, null, out, encoding);
+            if (StringUtils.isNotEmpty(position)) {
```

Comment:
I think it's better to evaluate against a default, where the default is set at the connect method in order to avoid having to call Integer.parseInt and StringUtils is not empty multiple times.

Reply:
Fixed in be664e04cdaea6376d91b834f294b25694dce0c6